provider/kubernetes/kubernetes.go

```diff
@@ -178,12 +178,34 @@ func (p *Provider) loadIngresses(k8sClient Client) (*types.Configuration, error)
 		}
 		templateObjects.TLS = append(templateObjects.TLS, tlsSection...)
 
+		backendPercentageWeightMap, err := getPercentageWeightMap(i)
+		if err != nil {
+			log.Errorf("Invalid yaml format for backend weight annotation of ingress %s/%s: %v", i.Namespace, i.Name, err)
+			continue
+		}
+
 		for _, r := range i.Spec.Rules {
 			if r.HTTP == nil {
 				log.Warn("Error in ingress: HTTP is nil")
 				continue
 			}
 
+			var leftFractionPercentage float64 = 1
```

Comment:
```go
leftFractionPercentage, leftFractionInstanceCount := getLeftFraction(k8sClient, r, servicesPercentageWeights, i)
```

```go
func getLeftFraction(k8sClient Client, rule extensionsv1beta1.IngressRule, servicesPercentageWeights map[string]*PercentageValue, i *extensionsv1beta1.Ingress) (float64, int) {
	var leftFractionPercentage float64 = 1
	leftFractionInstanceCount := 0

	for _, pa := range rule.HTTP.Paths {
		percentageWeight, found := servicesPercentageWeights[pa.Backend.ServiceName]
		if found {
			leftFractionPercentage -= percentageWeight.Float64()
			continue
		}

		endpoints, exist, err := k8sClient.GetEndpoints(i.Namespace, pa.Backend.ServiceName)
		if err == nil && exist {
			for _, subset := range endpoints.Subsets {
				leftFractionInstanceCount += len(subset.Addresses)
			}
		}
	}

	return leftFractionPercentage, leftFractionInstanceCount
}
```

morever, I think `leftFractionPercentage` must be a `PercentageValue` instead of `float64`.